apps/api/v2/src/modules/organizations/controllers/event-types/organizations-event-types.controller.ts

```diff
@@ -91,6 +94,30 @@ export class OrganizationsEventTypesController {
     };
   }
 
+  @Roles("TEAM_ADMIN")
+  @Post("/teams/:teamId/event-types/:eventTypeId/create-phone-call")
+  @UseGuards(ApiAuthGuard, IsOrgGuard, IsTeamInOrg, RolesGuard)
+  async createPhoneCall(
+    @Param("eventTypeId") eventTypeId: number,
+    @Param("orgId", ParseIntPipe) orgId: number,
+    @Body() body: CreatePhoneCallInput,
+    @GetUser() user: UserWithProfile
+  ): Promise<CreatePhoneCallOutput> {
+    const data = await handleCreatePhoneCall({
+      user: {
+        id: user.id,
+        timeZone: user.timeZone,
+        profile: { organization: { id: orgId } },
+      },
+      input: { ...body, eventTypeId },
```

Comment:
We are passing `eventTypeId` to `handleCreatePhoneCall` which has `TCreatePhoneCallSchema` for input, but `eventTypeId` does not exist on that schema.

Reply:
It has eventTypeId. check this code

```
const requiredFields = z.object({
  yourPhoneNumber: z.string().refine((val) => isValidPhoneNumber(val)),
  numberToCall: z.string().refine((val) => isValidPhoneNumber(val)),
  calApiKey: z.string().trim().min(1, {
    message: "Please enter CAL API Key",
  }),
  eventTypeId: z.number(),
  enabled: z.boolean().default(false),
  templateType: templateTypeEnum,
});

export const createPhoneCallSchema = requiredFields.merge(
  z.object({
    schedulerName: z.string().min(1).optional().nullable(),
    guestName: z
      .string()
      .optional()
      .transform((val) => {
        return !!val ? val : undefined;
      }),
    guestEmail: z
      .string()
      .optional()
      .transform((val) => {
        return !!val ? val : undefined;
      }),
    guestCompany: z
      .string()
      .optional()
      .transform((val) => {
        return !!val ? val : undefined;
      }),
    beginMessage: z.string().optional(),
    generalPrompt: z.string().optional(),
  })
);
```

Notice we are using merge